### PR TITLE
CoreData implementation

### PR DESCRIPTION
**Pull request by mduranx64**

### CODE REVIEW

File: FeedStoreChallenge/Feed Cache/CoreDataFeedStore.swift [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    }()
+    
+    private var context: NSManagedObjectContext {
+        return persistentContainer.viewContext
+    }
+    
+    public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
+        do {
+            try deleteCache()
+            completion(nil)
```
Comment (caiozullo) [inline FeedStoreChallenge/Feed Cache/CoreDataFeedStore.swift:106]: Missing `try context.save()`

File: Tests/FeedStoreChallengeTests.swift [hunk 2]
```diff
@@ -12,136 +12,112 @@ class FeedStoreChallengeTests: XCTestCase, FeedStoreSpecs {
...
+    private func undoStoreSideEffects() {
+        deleteStoreDataBase()
+    }
+    
+    private func deleteStoreDataBase() {
+        try? FileManager.default.removeItem(at: testSpecificStoreURL())
+    }
+    
+    private func testSpecificStoreURL() -> URL {
+        return cachesDirectory().appendingPathComponent("database")
```
Comment (caiozullo) [inline Tests/FeedStoreChallengeTests.swift:117]: You could use `/dev/null` to speed up the tests and avoid writing side-effects to disk.

Comment (caiozullo) [inline Tests/FeedStoreChallengeTests.swift:117]: You could use `/dev/null` to speed up the tests and avoid writing artifacts to disk.

File: FeedStoreChallenge/Feed Cache/CoreDataFeedStore.swift (hunk 1 above)
Comment (caiozullo) [inline FeedStoreChallenge/Feed Cache/CoreDataFeedStore.swift:106]: Missing `try context.save()` - you could check this behavior with an integration test.

File: FeedStoreChallenge/Feed Cache/CoreDataFeedStore.swift [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+//  FeedStoreChallenge
+//
+//  Created by Miguel Duran on 03-06-20.
+//  Copyright © 2020 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+public class CoreDataFeedStore: FeedStore {
```
Comment (caiozullo) [inline FeedStoreChallenge/Feed Cache/CoreDataFeedStore.swift:12]: At the moment the implementation works fine on the main thread. To make your implementation thread-safe, you can wrap the insert/retrieve/delete operations in a `context.perform` block:

https://developer.apple.com/documentation/coredata/nsmanagedobjectcontext/1506578-perform

File: FeedStoreChallenge/Feed Cache/CoreDataFeedStore.swift (hunk 1 above)
Comment (mduranx64) [reply FeedStoreChallenge/Feed Cache/CoreDataFeedStore.swift]: I added an integration test and it fail when `try context.save()` is missing.

File: Tests/FeedStoreChallengeTests.swift (hunk 2 above)
Comment (mduranx64) [reply Tests/FeedStoreChallengeTests.swift]: I change the tests to use `/dev/null`.

File: FeedStoreChallenge/Feed Cache/CoreDataFeedStore.swift (hunk 3 above)
Comment (mduranx64) [reply FeedStoreChallenge/Feed Cache/CoreDataFeedStore.swift:12]: Thank you for the detailed review
I change the implementation to use `context.perfom` block

**Comment by caiozullo** [comment]
Nice one!

**Pull request closed without merging**